packages/otelbin/src/components/validation-type/ValidationType.tsx

```diff
@@ -1,38 +1,86 @@
 // SPDX-FileCopyrightText: 2023 Dash0 Inc.
 // SPDX-License-Identifier: Apache-2.0
 
-import { useState } from "react";
+import { useEffect, useMemo, useState } from "react";
 import { Button } from "../button";
 import { Popover, PopoverContent, PopoverTrigger } from "../popover";
 import Down from "./../assets/svg/down.svg";
 import ValidationTypeContent from "./ValidationTypeContent";
+import { useDistributions } from "../validation/useDistributions";
+import { useUrlState } from "~/lib/urlState/client/useUrlState";
+import { distroBinding, distroVersionBinding } from "../validation/binding";
 
 export interface ICurrentValidation {
-	provider: string;
-	version: string;
-	distro: string;
+	title: {
+		provider: string;
+		version: string;
+		distro: string;
+	};
+	selected: {
+		provider: string;
+		distro: string;
+		version: string;
+	};
 }
 
 export default function ValidationType() {
-	const [current, setCurrent] = useState<ICurrentValidation>({
-		provider: "Browser-only",
-		version: "",
-		distro: "",
+	const [{ distro, distroVersion }] = useUrlState([distroBinding, distroVersionBinding]);
+	const [currentDistro, setCurrentDistro] = useState<ICurrentValidation>({
```

Comment:
**Question:** Why do we need this state? 

I would expect that the combination of URL state and the result of `useDistributions` is sufficient to generate the right label. This would also avoid the need for `useMemo` and `useEffect`.

Reply:
Exactly today I tried the approach without useState, but at final step, without useState and useEffect, for the first state change, useUrlState couldn't change the whole state of the app, (ex: display current flag, or button title). If you're sure useUrlState is enough, I'll try again.